regparser/tree/reg_text.py

```diff
@@ -84,6 +85,95 @@ def build_subpart(text, part):
         "", [], label, subpart_title, node_type=struct.Node.SUBPART)
 
 
+def subjgrp_label(starting_text, letter_list):
+    words = starting_text.split(" ")
+    candidate_text = ""
+    suffixes = [""] + [_ for _ in string.lowercase[1:]]
+    if len(words) == 1:
+        """
+        E.g. if the word is "Penalties" the progression is:
+
+        Pe
+        Pe.
+        Pen
+        Pen.
+        Pena
+        Pena.
+        <etc.>
+        Penalties.
+        Penalties-b.
+        Penalties-c.
+        <etc.>
+        """
+        word = words[0]
+        terminators = ("", ".")
+        suffix_pos, terminator_pos = 0, 0
+        pos = min([2, len(word)])
+        while candidate_text == "" or candidate_text in letter_list:
+            suffix = '-%s' % suffixes[suffix_pos] if suffix_pos else ''
+            candidate_text = '%s%s%s' % (word[:pos],
+                                         terminators[terminator_pos], suffix)
+            if not terminator_pos:
+                terminator_pos = 1
+                continue
+            else:
+                terminator_pos = 0
+            if pos < len(word):
+                pos = pos + 1
+            else:
+                suffix_pos = suffix_pos + 1
```

Comment:
Might be worth rewriting as an `if-elif-else` to avoid the `continue`
```python
if not terminator_pos:
    terminator_pos = 1
elif pos < len(word):
    terminator_pos = 0
    pos += 1
else:
    terminator_pos = 0
    suffix_pos += 1
```